GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -0,0 +1,38 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public partial class ProductUpgrader
+    {
+        public static IProductUpgrader CreateUpgrader(ITracer tracer, out string error)
+        {
+            IProductUpgrader upgrader;
+            bool isEnabled;
+            bool isConfigured;
+            error = string.Empty;
+            
+            if ((upgrader = GitHubUpgrader.Create(tracer, out isEnabled, out isConfigured)) != null)

```

Comment:
this assignment inside the `if` is a bit messy. Pull the assignment out, then check `upgrader != null` in the `if` condition.